src/core.js

```diff
@@ -1794,15 +1828,6 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
 
     instance.runHooks('afterCellMetaReset');
 
-    if (isDefined(settings.className)) {
-      if (GridSettings.prototype.className) {
-        removeClass(instance.rootElement, GridSettings.prototype.className);
-      }
-      if (settings.className) {
-        addClass(instance.rootElement, settings.className);
-      }
-    }
```

Comment:
Maybe here we could extend the algorithm to adjust `className` and `tableClassName`?

Reply:
But both settings we may set at the same time so for convenience, I choose a function.